src/core/thread/network_data.hpp

```diff
@@ -427,14 +437,18 @@ class NetworkData : public InstanceLocator
      * @param[in]  aEnterpriseNumber  Enterprise Number.
      * @param[in]  aServiceData       A pointer to a Service Data.
      * @param[in]  aServiceDataLength The Service Data length pointed to by @p aServiceData.
+     * @param[in]  aServiceMatchMode  The Service Data match mode. @sa ServiceDataMatchMode
```

Comment:
nit: (other instances below)
```suggestion
     * @param[in]  aServiceMatchMode  The Service Data match mode.
```